src/microvm/src/vmm/microvm/mod.rs

```diff
@@ -85,10 +89,7 @@ impl Vmm {
         let (vm_tx, gateway_rx) = mpsc::channel::<Message>();
         let (gateway_tx, memory_thread_rx) = mpsc::channel::<Message>();
         let (memory_thread_tx, vm_rx) = mpsc::channel::<Message>();
-
-        // Spawn I/O thread.
-        let _io_thread: Option<JoinHandle<Result<()>>> =
-            gateway_conn.map(|conn| IoThread::spawn(conn, gateway_rx, gateway_tx.clone()));
+        let (paused_tx, paused_rx) = mpsc::channel::<Message>();
```

Comment:
You could  create `event_tx` and `event_rx` here as well.

Reply:
> This choice is cosmetic.
I disagree. In one you end up with forcing ownership transfers. Let's go with the solution where you forgo the concept of a `VirtualProcessorHandle`.

As a side note, I believe that the best long-term architectural decision would be to spawn a "vcu thread" (which is currently running as the "main thread"), and turn the "iothread" into the "main thread".

Reply:
I would say that we are almost on the same page. I pushed some changes to this branch as I found the ownership for the VMM thread handle (previously vCPU thread) messy.